Lesson1

slexing opened the pull request:
Home work for Lesson1 with 4 Execrises and some stuff like Methods, Classes, Exceptions and etc

slexing committed:
Delete ConsoleApp1 directory

slexing committed:
lesson1

NaGriS reviewed (commented):
There is nothing to review, there is only file renames

NaGriS commented:
Looks like you committed it to master, then deleted, and now git couldn't see changes.
Need to be fixed somehow, if there would be a problem - we could do it together

Commit:
Merge branch 'master' into Lesson1

# Conflicts:
#	Lesson1/.gitignore
#	Lesson1/Exercise1/Exercise1.csproj
#	Lesson1/Exercise1/Program.cs
#	Lesson1/Exercise1/Properties/AssemblyInfo.cs
#	Lesson1/Exercise1/app.config
#	Lesson1/Exercise2/Exercise2.csproj
#	Lesson1/Exercise2/Program.cs
#	Lesson1/Exercise2/Properties/AssemblyInfo.cs
#	Lesson1/Exercise2/Shape.cs
#	Lesson1/Exercise2/app.config
#	Lesson1/Exercise3/App.config
#	Lesson1/Exercise3/Exercise3.csproj
#	Lesson1/Exercise3/Program.cs
#	Lesson1/Exercise3/Properties/AssemblyInfo.cs
#	Lesson1/Exercise4/Exercise4.csproj
#	Lesson1/Exercise4/Program.cs
#	Lesson1/First Homework.sln

NaGriS commented on Lesson1/Exercise2/Program.cs:
```diff
@@ -0,0 +1,35 @@
+                if (shape.ValidateSides(side1, side2, side3))
+                {
+                    double d = shape.CalculateDiagonal();
+                    Console.WriteLine($"Диагональ пареллепипеда равна: {d}");
```
Using russian and english languages in one program - bad practise

NaGriS commented on Lesson1/Exercise2/Shape.cs:
```diff
@@ -0,0 +1,60 @@
+            }
+        }
+        //Method for zero check
+        private void ThrowArgumentException(int side)
```
Bad method name - it is validating that side is not zero and below

NaGriS commented on Lesson1/Exercise2/Shape.cs:
```diff
@@ -0,0 +1,60 @@
+
+
+
+
```
Good use of classes, nice

NaGriS commented on Lesson1/Exercise3/Program.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Введите первую строку:");
```
Russian?

NaGriS commented on Lesson1/Exercise3/Program.cs:
```diff
@@ -0,0 +1,37 @@
+            Console.WriteLine("Введите третью строку:");
+            String third = Console.ReadLine();
+
+            String firstLowerCase = first.ToLower();
```
As it said in homework it should be : "ПРОПИСНОЙ" format, and you have "строчный" format

NaGriS commented on Lesson1/Exercise3/Program.cs:
```diff
@@ -0,0 +1,37 @@
+            String thirdLowerCase = third.ToLower();
+
+            //Method for startWith First to Second
+            bool CompareFirstSecond()
```
You've implemented method inside method (very bad readability) and such method doing only one action that could be performed by system methods
For what ?

NaGriS commented on Lesson1/Exercise4/Program.cs:
```diff
@@ -0,0 +1,25 @@
+            Console.WriteLine("Введите вторую строку");
+            string second = Console.ReadLine();
+            Console.WriteLine("Введите символ");
+            char symbol = Console.ReadKey().KeyChar;
```
Nice job there!

NaGriS commented on Lesson1/Exercise4/Program.cs:
```diff
@@ -0,0 +1,25 @@
+            var sum = first +" "+second;
+            var index = sum.LastIndexOf(symbol);
+
+            Console.WriteLine($"Concatenation string is {sum}");
```
Again russian and english in one program + misspell in english word

NaGriS reviewed (commented):
Almost ready for approve, small fixes needed

slexing committed:
Correction for Lesson1 exercises
1) Remove russian wordings
2) Exercise3: Remove methods
3) Exercise3: Change toLower -> toUpper
4) Exercise2: Change method name ThrowArgumentException -> SideZeroValidate

slexing committed:
Delete commented code

NaGriS reviewed (approved):
Great
